llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp

```diff
@@ -726,7 +726,8 @@ std::optional<bool> ZeroOverheadLoop::createTripCountGreaterCondition(
 
 void ZeroOverheadLoop::adjustTripCount(int TripCountAdjust) {
   LLVM_DEBUG(dbgs() << "TripCountAdjust =  " << TripCountAdjust << "\n");
-  if (DefTripCount->getOperand(1).isImm()) {
+  if (DefTripCount->getOperand(1).isImm() &&
```

Comment:
This conflicts with https://github.com/Xilinx/llvm-aie/pull/146. As agreed with @martien-de-jong  his solution will cover this case, so this is just a temporary fix.